src/api/course/index.ts

```diff
@@ -320,19 +321,55 @@ export class CourseApi {
     return _.get((this.app.signalk as any).self, 'navigation.position')
   }
 
-  private validateCourseInfo(info: CourseInfo) {
+  private async validateCourseInfo(info: CourseInfo) {
     if (
-      typeof info.activeRoute !== 'undefined' &&
-      typeof info.nextPoint !== 'undefined' &&
-      typeof info.previousPoint !== 'undefined'
+      typeof info.activeRoute === 'undefined' ||
+      typeof info.nextPoint === 'undefined' ||
+      typeof info.previousPoint === 'undefined'
```

Comment:
I don't think this is correct - this says that if any one of these is undefined when the route is invalid, but the tests below cater for the case where `!info.activeRoute && info.nextPoint?.href` so that branch will never fire.

Reply:
The Stored c onfiguration is expected to have all of the keys: `activeRoute`,`nextPoint`,`previousPoint`.
So if one is not defined then the stored config is considered invalid, so no further processing is required.

If the destination is a waypoint / position (not a route) then `activeRoute` will be null (not undefined)

Reply:
🤦 thanks. I refactored this a bit more for clarity of intent, please check.